### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by betolink**
A PR to include the new Notebooks for IceFlow, they are not in conflict with the rest of the tutorials and were reviewed by Amy F. and Nicholas K.

### CODE REVIEW

File: notebooks/iceflow/3_dataviz.ipynb [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    "df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Vaex \n",
+    "\n",
+    "* Using VAEX to visualize big data frames.\n"
```
Comment (asteiker) [inline notebooks/iceflow/3_dataviz.ipynb:76]: See comments from intro notebook: I get errors when I use vaex from my IceFlow output.

File: notebooks/iceflow/3_dataviz.ipynb [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    "\n",
+    "filepath = 'data/twaties-test-GLAH06-2000-2010.h5'\n",
+    "df_key = ifp.get_common_dictionary('GLASS')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading the data with H5PY"
```
Comment (asteiker) [inline notebooks/iceflow/3_dataviz.ipynb:51]: I received this error from the LVIS output from iceflow: 

`AttributeError: 'NoneType' object has no attribute 'describe'`

File: notebooks/iceflow/3_dataviz.ipynb [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Time Series\n",
+    "\n",
+    "After harmonizing data from Pre-IB, IS1, IceBridge, IS2 we should be able to grid and or concatenate the dataframes and start building real analyses.\n",
+    "\n",
+    "Ideas: \n",
```
Comment (asteiker) [inline notebooks/iceflow/3_dataviz.ipynb:205]: Is this still a todo for the notebook? These look like internal notes, not something we should publish.

File: notebooks/iceflow/0_introduction.ipynb [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+   "source": [
+    "orders = ui.place_data_orders(params=my_params)\n",
+    "orders"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Downloading the data\n",
```
Comment (asteiker) [inline notebooks/iceflow/0_introduction.ipynb:326]: The output naming convention is not ideal - It's really hard to know what output file came from what data set. Maybe this is a Hermes enhancement? The output files should retain the input file name for provenance.

File: notebooks/itslive/itslive.ipynb [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    "\n",
+    "* Coverage: All land ice\n",
+    "* Date range: 1985-present\n",
+    "* Resolution: 240m\n",
+    "* Scene-pair separation: 6 to 546 days\n",
+    "\n",
+    "\n",
+    "---\n",
+    "\n",
+    "* If you want to query our API directly using  your own software here is the OpenApi endpoint https://staging.nsidc.org/apps/itslive-search/docs\n",
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:25]: This is ready to merge with the API still in staging? Does this need to point to a PROD endpoint instead?

File: notebooks/itslive/itslive.ipynb [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    "* For questions about this notebook and the dataset please contact users services at uso@nsidc.org\n"
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:26]: change to nsidc@nsidc.org

File: notebooks/itslive/itslive.ipynb [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#2: We build the parameters to query the ITS_LIVE Search API, we get the time coverage for our selected area\n",
+    "params = ui.build_params()\n",
+    "print(f'current parameters: {params}')\n",
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:51]: I get this error:

`
You need to select an area using the box tool or the polygon drawing tool
current parameters: None`

But there are no bbox or polygon options in the map - just "circle marker" option.

File: notebooks/itslive/itslive.ipynb [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    "timeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#3: Now we are going to get the velocity pair urls, this does not download the files yet just their location\n",
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:66]: I don't understand the difference between this step and the next step to filter granules. Extra markdown guidance would be helpful.

File: notebooks/itslive/itslive.ipynb [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+   "source": [
+    "#4: This will query AWS(where the granules are stored) so we know the total size of our first N granules\n",
+    "# This may take some time, try reducing the selected area or constraining the other parameters to download a reasonable number of granules.\n",
+    "url_list = [url['url'] for url in urls]\n",
+    "# urls\n",
+    "filtered_urls = ui.filter_urls(url_list, max_files_per_year=5, months=[3,4,5,6,7])\n",
+    "\n",
+    "# max_granules = 100\n",
+    "# sizes = ui.calculate_file_sizes(filtered_urls, max_granules)\n",
+    "# total_zise = round(sum(sizes)/1024,2)\n",
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:97]: Do you want to keep this commented?

File: notebooks/itslive/itslive.ipynb [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Downloading the data\n",
+    "**Now that we have our list of data granules we can download them from AWS.**\n",
+    "\n",
+    "If this notebook is running inside AWS we could load the granules into a Dask cluster and reduce our processing times and costs.\n",
+    "Let's get some coffee, some data requests are in the Gigabytes realm and may take a little while to be processed. \n",
+    "Once that your status URL says is completed we can grab the HDF5 data file using the URL on the same response!"
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:112]: I never saw a status outputted here but everything completed.

File: notebooks/itslive/itslive.ipynb [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+    "import xarray as xr\n",
+    "import pyproj\n",
+    "import warnings\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "from datetime import datetime\n",
+    "from shapely.geometry import Polygon\n",
+    "warnings.filterwarnings('ignore')\n",
+    "centroid = ui.dc.last_draw['geometry']['coordinates']\n",
+    "# coord = [-49.59321, 69.210579]\n",
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:150]: Keep?

File: notebooks/itslive/itslive.ipynb [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+    "#loads an array of xarray datasets from the nc files\n",
+    "velocity_pairs = ui.load_velocity_pairs('data')\n",
```
Comment (asteiker) [inline notebooks/itslive/itslive.ipynb:152]: Is 'data' supposed to be a single file path? I tried this on a few of my output files and my velocities were empty. Not sure what I shuold be doing here.

**Review by asteiker — CHANGES_REQUESTED** [review]
Lots of comments, Luis, but a lot of them are minor typos and markdown adjustments. However I had quite a few errors on many of the notebooks and these need to be addressed before merging. We could always separate the itslive changes if you need iceflow merged first (or visa versa).

File: notebooks/iceflow/0_introduction.ipynb [hunk 13]
```diff
@@ -0,0 +1,725 @@
...
+    "IceBridge flights were generally conducted in **March-May over Greenland and in October-November over Antarctica**.\n",
+    "\n",
+    "### **ICESat 2**\n",
+    "\n",
+    "The ICESat-2 mission is designed to provide elevation data needed to determine ice sheet mass balance as well as vegetation canopy information. It will provide topographic measurements of cities, lakes and reservoirs, oceans and land surfaces around the globe. The sole instrument on ICESat-2 is the Advanced Topographic Laser Altimeter System (ATLAS), a space-based Lidar. It was designed and built at Goddard Space Flight Center, with the laser generation and detection systems provided by Fibertek. ATLAS measures the travel time of laser photons from the satellite to Earth and back; computer programs use the travel time from multiple pulses to determine elevation.\n",
+    "\n",
+    "You can go to NSIDC's landing page for a complete list of IceSat 2 datasets and their documentation: [ICESat-2 Data Sets at NSIDC](https://nsidc.org/data/icesat-2/data-sets)\n",
+    "\n",
+    "<p align=\"center\">\n",
+    "<img style=\"align: center;\" width=\"80%\" src='./img/iceflow-coverage.jpg'/>\n",
```
[the remark this answers is not in the document]
Comment (betolink) [reply notebooks/iceflow/0_introduction.ipynb:95]: The image is now updated, the ICESat website says it was decommissioned on 2010.

File: notebooks/itslive/itslive.ipynb (hunk 10 above)
Comment (betolink) [reply notebooks/itslive/itslive.ipynb]: I just changed the text

File: notebooks/itslive/itslive.ipynb (hunk 12 above)
Comment (betolink) [reply notebooks/itslive/itslive.ipynb:153]: data is the name of the directory you want to load the velocity granules from. Clarified in a comment.

**Pull request merged**